pinot-common/src/main/java/org/apache/pinot/pql/parsers/PinotQuery2BrokerRequestConverter.java

```diff
@@ -0,0 +1,235 @@
+      brokerRequest.setSelections(selection);
+    }
+
+    if (aggregationInfoList != null && aggregationInfoList.size() > 0) {
+      brokerRequest.setAggregationsInfo(aggregationInfoList);
+    }
+
+    //Handle group by
+    List<Expression> groupByList = pinotQuery.getGroupByList();
+    if (groupByList != null && groupByList.size() > 0) {
+
+      groupBy = new GroupBy();
+      for (Expression expression : groupByList) {
+        String expressionStr = standardizeExpression(expression);
+        groupBy.addToExpressions(expressionStr);
+      }
+      groupBy.setTopN(pinotQuery.getLimit());
+    }
+
+    //Query Type
+    QueryType queryType = new QueryType();
+    if (aggregationInfoList != null && aggregationInfoList.size() > 0) {
+      if (groupBy != null) {
+        queryType.setHasGroup_by(true);
+      } else {
+        queryType.setHasAggregation(true);
+      }
+    } else {
+      queryType.setHasSelection(true);
+    }
+    brokerRequest.setQueryType(queryType);
+
+    //TODO: these should not be part of the query?
+    //brokerRequest.setEnableTrace();
+    //brokerRequest.setDebugOptions();
+    //brokerRequest.setQueryOptions();
+    //brokerRequest.setBucketHashKey();
+    //brokerRequest.setDuration();
+
+    return brokerRequest;
+  }
+
+  private String standardizeExpression(Expression expression) {
+    switch (expression.getType()) {
+      case LITERAL:
+        return expression.getLiteral().getValue();
+      case IDENTIFIER:
+        return expression.getIdentifier().getName();
+      case FUNCTION:
+        Function functionCall = expression.getFunctionCall();
+        StringBuilder sb = new StringBuilder();
+        sb.append(functionCall.getOperator());
+        sb.append("(");
+        for (Expression operand : functionCall.getOperands()) {
+          sb.append(standardizeExpression(operand));
+        }
+        sb.append(")");
+        return sb.toString();
+      default:
+        throw new UnsupportedOperationException("Unknown Expression type: " + expression.getType());
+    }
+  }
+
+  private AggregationInfo buildAggregationInfo(Function function) {
+    List<Expression> operands = function.getOperands();
+    if (operands == null || operands.size() != 1) {
+      throw new Pql2CompilationException(
+          "Aggregation function" + function.getOperator() + " expects 1 argument. found: "
+              + operands);
+
+    }
+    String functionName = function.getOperator();
+    String columnName;
+    if (functionName.equalsIgnoreCase("count")) {
+      columnName = "*";
+    } else {
+      Expression functionParam = operands.get(0);
+
+      switch (functionParam.getType()) {
+        case LITERAL:
+          columnName = functionParam.getLiteral().getValue();
+          break;
+        case IDENTIFIER:
+          columnName = functionParam.getIdentifier().getName();
+          break;
+        case FUNCTION:
+        default:
+          throw new UnsupportedOperationException(
```

Comment:
I was under the impression that function is supported within aggregations (added by Jackie)?

Reply:
hmm, will check with him

Reply:
could not get hold of Jack. Will fix it when we test this for backward compatibility